src/main/java/com/imsweb/geocoder/GeocodeOutput.java

```diff
@@ -467,11 +467,14 @@ public void setPenaltyCodeSummary(String penaltyCodeSummary) {
                         result.setNaaccrCensusTractCertaintyName(value(parts[115]));
 
                         // test if there are any census tracts returned
-                        if (parts.length > 119) {
+                        if (parts.length > 130) {
                             addCensus(result, parts, 1990, 116);
                             addCensus(result, parts, 2000, 127);
                             addCensus(result, parts, 2010, 138);
                         }
+                        else if (parts.length > 119) {
+                            addCensus(result, parts, 2010, 116);
```

Comment:
Micro Match Status, Penalty Code, and Penalty Code Summary will be ignored when only one census tract is returned. Also, why do we check if the length is greater than 130 when we know will be using 149 parts? I'd like this code to as safe as possible.

Do you think something like this would work?
```java
// test if there are any census tracts returned
int nextPosition;
if (parts.length > 148) {
    addCensus(result, parts, 1990, 116);
    addCensus(result, parts, 2000, 127);
    addCensus(result, parts, 2010, 138);
    nextPosition = 149;
}
else if (parts.length > 126) {
    addCensus(result, parts, *year parameter*, 116);
    nextPosition = 127;
}
else
    nextPosition = 116;

if (parts.length > nextPosition + 2) {
    result.setMicroMatchStatus(value(parts[nextPosition]));
    result.setPenaltyCode(value(parts[nextPosition + 1]));
    result.setPenaltyCodeSummary(value(parts[nextPosition + 2]));
}
```